doc/api/assert.md

```diff
@@ -2548,6 +2548,241 @@ assert.throws(throwingFirst, /Second$/);
 Due to the confusing error-prone notation, avoid a string as the second
 argument.
 
+## `assert.deepMatch(actual, expected[, message])`
```

Comment:
The naming here might be confusing given that `assert.match(...)` exists and uses regexes but this `deepMatch()` has nothing to do with regexes.

It's a bit more verbose both something like `assert.partialDeepEqual()` and `assert.partialDeepStrictEqual()` might be better / less confusing?

Reply:
@jasnell I just followed the most "approved" comment in the original PR and implemented it :) 

https://github.com/nodejs/node/issues/50399#issuecomment-2296531960

Reply:
FWIW, +1 regarding the naming semantics. My only alternative suggestion would be `deepEqualSubset()`, which is potentially a bit more intuitive than "partial" (although ES2025 will introduce the Set-specific notion of subset matching, which could itself be a source of confusion).

Reply:
@Renegade334 to be completely honest I like the "partial" more, I have the feeling it is a little more immediate to understand what the method does.  
But, I would like to have a final decision here 😄 is there any way we can tag people to take a final decision here?